### PR TITLE
6 assign and use custom variables

### PR DESCRIPTION
**Pull request by henrikprangelNG**
* Implement assign steps for variable assignment. 
* Implement ability to use assigned variables in other steps
* Add initial version of scripting _(currently utilizing java scripting API to script with js - this will be more thoroughly tested, and replaced if needed, in coming tasks to validate if this solution is secure)_
* Refactor FileUtils to use newer nio instead of io
* general code cleanup

### CODE REVIEW

File: src/main/java/ee/buerokratt/ruuter/controller/ConfigurationController.java [hunk 1]
```diff
@@ -9,16 +9,18 @@
...
+import static org.springframework.web.bind.annotation.RequestMethod.POST;
 
 @RestController
 @RequiredArgsConstructor
 public class ConfigurationController {
     private final ConfigurationService configurationService;
 
-    @RequestMapping(value = "/{configuration}", method = {RequestMethod.GET, RequestMethod.POST})
-    public ResponseEntity<RuuterResponse> queryConfiguration(@PathVariable("configuration") String code,
+    @RequestMapping(value = "/{configuration}", method = {GET, POST})
```
Comment (turnerrainer) [inline src/main/java/ee/buerokratt/ruuter/controller/ConfigurationController.java:20]: Method types should be defined as setting variables.

https://github.com/orgs/buerokratt/projects/5/views/3

File: src/main/java/ee/buerokratt/ruuter/helper/ConfigurationMappingHelper.java [hunk 2]
```diff
@@ -1,55 +1,56 @@
...
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 import ee.buerokratt.ruuter.domain.AssignStep;
 import ee.buerokratt.ruuter.domain.ConfigurationStep;
 import ee.buerokratt.ruuter.domain.HttpStep;
 import ee.buerokratt.ruuter.domain.ReturnStep;
-import ee.buerokratt.ruuter.service.exception.InvalidConfigurationException;
-import ee.buerokratt.ruuter.service.exception.InvalidConfigurationStepException;
+import ee.buerokratt.ruuter.helper.exception.InvalidConfigurationException;
```
Comment (turnerrainer) [inline src/main/java/ee/buerokratt/ruuter/helper/ConfigurationMappingHelper.java:11]: I'd say the previous version (`service`, not `helper`) was correct. Helpers should be common classes to create UUIDs etc. Functionalities that are not Ruuter-specific.

The same goes throughout this pull request but I only added 1 comment about it.

**Review by turnerrainer — CHANGES_REQUESTED** [review]
Define allowed method types as setting variables.

File: src/main/java/ee/buerokratt/ruuter/controller/ConfigurationController.java (hunk 1 above)
Comment (turnerrainer) [reply src/main/java/ee/buerokratt/ruuter/controller/ConfigurationController.java:20]: Create a new task for it.

Comment (henrikprangelNG) [reply src/main/java/ee/buerokratt/ruuter/controller/ConfigurationController.java:20]: Crated task for this spesific issue:
https://github.com/buerokratt/Ruuter/issues/41

**Review by turnerrainer — APPROVED** [review]
Defining allowed method types will be done via [another task](https://github.com/buerokratt/Ruuter/issues/41)

**Review by turnerrainer — APPROVED** [review]

**Pull request merged**